client-runtime/src/main/java/com/microsoft/rest/v2/http/FlowableFileStream.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.http;
+
+import io.reactivex.Flowable;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.CompletionHandler;
+import java.util.Arrays;
+
+/**
+ * Exposes an AsynchronousFileChannel as a Flowable.
+ *
+ * Does not close the AsynchronousFileChannel after completion.
+ */
+public class FlowableFileStream extends Flowable<byte[]> {
+    private static final int CHUNK_SIZE = 8192;
+    private final AsynchronousFileChannel fileChannel;
+    private final long offset;
+    private final long length;
+
+    /**
+     * Creates a FlowableFileStream which reads a file starting from an offset and length.
+     * @param fileChannel The AsynchronousFileChannel to read file content from.
+     * @param offset The offset in the file to begin reading.
+     * @param length The number of bytes of data to read from the file.
+     */
+    public FlowableFileStream(AsynchronousFileChannel fileChannel, long offset, long length) {
+        this.fileChannel = fileChannel;
+        this.offset = offset;
+        this.length = length;
+    }
+
+    /**
+     * Creates a FlowableFileStream which reads an entire file.
+     * @param fileChannel The AsynchronousFileChannel to read content from
+     * @throws IOException if an exception occurs when determining file size
+     */
+    public FlowableFileStream(AsynchronousFileChannel fileChannel) throws IOException {
+        this(fileChannel, 0, fileChannel.size());
+    }
+
+    @Override
+    protected void subscribeActual(final Subscriber<? super byte[]> subscriber) {
+        subscriber.onSubscribe(new Subscription() {
+            final ByteBuffer innerBuf = ByteBuffer.wrap(new byte[CHUNK_SIZE]);
+            boolean canceled = false;
+            long chunksRequested = 0;
+            long position = offset;
+
+            @Override
+            public void request(long n) {
+                chunksRequested += n;
+                readSegmentsAsync();
+            }
+
+            void readSegmentsAsync() {
+                fileChannel.read(innerBuf, position, null, onReadComplete);
+            }
+
+            final CompletionHandler<Integer, Object> onReadComplete = new CompletionHandler<Integer, Object>() {
```

Comment:
Any reason this is package private and not private?

Reply:
`private` helps to convey intent, but non-overriding members of anonymous classes aren't accessible from outside. Might as well have it, though.